lib/extend-default-visitor-keys.js

```diff
@@ -0,0 +1,21 @@
+"use strict";
+
+var BABEL_VISITOR_KEYS = require("babel-types").VISITOR_KEYS;
+
+/**
+ * Generate extended visitor keys, based on the default keys from ESLint and the
+ * VISITOR_KEYS from babel-types
+ * 
+ * @param {Object} defaultVisitorKeys The default visitor keys from the user's ESLint
+ * @returns {Object} a final copy of visitor keys, extended using those from babel-types
+ */
+module.exports = function extendDefaultVisitorKeys(defaultVisitorKeys) {
+  var extendedVisitorKeys = Object.assign(
+    {},
+    defaultVisitorKeys,
+    BABEL_VISITOR_KEYS
+  );
+  extendedVisitorKeys.MethodDefinition.push("decorators");
+  extendedVisitorKeys.Property.push("decorators");
```

Comment:
It looks like this still monkeypatches the default visitor keys from ESLint. Since `extendedVisitorKeys` is only a shallow clone, the `extendedVisitorKeys.MethodDefinition` array is still shared with the array from ESLint.